BitPayLib/class-bitpayipnprocess.php

```diff
@@ -21,6 +22,8 @@ class BitPayIpnProcess {
 
 	use WpDbHelper;
 
+	private const FINAL_WC_ORDER_STATUSES = array( 'wc-refunded', 'wc-cancelled', 'wc-failed' );
```

Comment:
@shaunek-hero we hesitate to make `completed` a final status. The reasoning is that someone could still perform a return after they receive their product.

This is from Woo's [documentation](https://woo.com/document/managing-orders/#visual-diagram-to-illustrate-order-statuses):

![image](https://github.com/bitpay/bitpay-checkout-for-woocommerce/assets/123703/d7679d94-add6-4c15-84d5-f91417532169)

Do you mean that you're recommending updating the status once complete would only ever be manual?

Reply:
Hey @shaunek-hero we've thought about this a bit, and here's where we are. We think that this should be an admin option: "Allow refund IPN to change status when order is complete".

There are two cases:
* Some merchants might want to create a refund in their BitPay dashboard and have it flow through to the shop, changing the status. This would satisfy returns after purchase.
* Other merchants might not want the status to change if the order is complete, so they can manually change it.

We only listen for the `refund_success` status, which is after the refund request flow is done / approved / paid out.